cmake/list_append_keys_from.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
+#
+macro(list_append_keys_from listname source)
+  foreach(_item ${${source}})
+      set(_found "")
+      string(REGEX REPLACE "^([^=]+)(=.*)$?" "^\\1(=.*)?$" _expr ${_item})
+      foreach(_elem ${${listname}})
+        if(_elem STREQUAL _item)
+          set(_found TRUE)
+          break()
+        elseif(_elem MATCHES ${_expr})
+          set(_found TRUE)
+          message(WARNING "could not add '${_item}' (from '${source}'), already contained: '${_elem}'")
```

Comment:
This warning would also appear if the two values `_item` and `_elem` are identical which doesn't make much sense.